utils.go

```diff
@@ -156,6 +77,65 @@ func LoadConfig(profile string, configDir string) (config *Config, err error) {
 	return config, nil
 }
 
+func GetConsulClient(serviceName string, profile string, config *Config) (*registry.ConsulClient, error) {
```

Comment:
this one not just returns consul client but also does many things with registry. IMHO: logic is too complicated. We have registryClient, consulClient and a bit inconsistency between function names and logic in it. As for me we should:

1. Check if we could connect to consul.
2. If yes: return consul client else return proper error message
3. register service and health check with consul client
4. check key value pairs
5. get config from consul

Reply:
Actually I want to continue integrate this package until we can direct access this package.
https://github.com/edgexfoundry/edgex-go/tree/master/internal/pkg/consul
Then we can do refactor according this https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/command/init.go#L100

Reply:
you can. IMHO we need to keep things granular: if function called GetConsulClient it should returns only consul client and doesn't do any other changes implicitly, for such case we can do next:

in serviceStart:
1. get consul client
2. register service with consul client if it's not
3. register health with consul client check if it's not
4. check key-value with consul client